modules/linear_boltzmann_solvers/response_evaluator/response_evaluator.cc

```diff
@@ -0,0 +1,468 @@
+  {
+    for (const auto& cell : grid.local_cells)
+    {
+      const auto& cell_mapping = discretization.GetCellMapping(cell);
+      const auto& transport_view = transport_views[cell.local_id_];
+      const auto& fe_values = unit_cell_matrices[cell.local_id_];
+      const auto num_cell_nodes = cell_mapping.NumNodes();
+
+      if (material_sources_.count(cell.material_id_) > 0)
+      {
+        const auto& src = material_sources_.at(cell.material_id_);
+        for (size_t i = 0; i < num_cell_nodes; ++i)
+        {
+          const auto dof_map = transport_view.MapDOF(i, 0, 0);
+          const auto& V_i = fe_values.intV_shapeI[i];
+          for (size_t g = 0; g < num_groups; ++g)
+            local_response += src[g] * phi_dagger[dof_map + g] * V_i;
+        }
+      }
+    } // for cell
+  }   // if material sources
+
+  // Boundary sources
+  if (not boundary_sources_.empty())
+  {
+    size_t gs = 0;
+    for (const auto& groupset : lbs_solver_.Groupsets())
+    {
+      const auto& uk_man = groupset.psi_uk_man_;
+      const auto& quadrature = groupset.quadrature_;
+      const auto& num_gs_angles = quadrature->omegas_.size();
+      const auto& num_gs_groups = groupset.groups_.size();
+
+      for (const auto& cell : grid.local_cells)
+      {
+        const auto& cell_mapping = discretization.GetCellMapping(cell);
+        const auto& fe_values = unit_cell_matrices[cell.local_id_];
+
+        size_t f = 0;
+        for (const auto& face : cell.faces_)
+        {
+          if (not face.has_neighbor_ and boundary_sources_.count(face.neighbor_id_) > 0)
+          {
+            const auto bndry_id = face.neighbor_id_;
+            const auto num_face_nodes = cell_mapping.NumFaceNodes(f);
+            for (size_t fi = 0; fi < num_face_nodes; ++fi)
+            {
+              const auto i = cell_mapping.MapFaceNode(f, fi);
+              const auto& node = grid.vertices[cell.vertex_ids_[i]];
+              const auto& intF_shapeI = fe_values.intS_shapeI[f][i];
+
+              const auto psi_bndry = EvaluateBoundaryCondition(bndry_id, node, groupset);
+
+              for (size_t n = 0; n < num_gs_angles; ++n)
+              {
+                const auto& omega = quadrature->omegas_[n];
+                const auto mu = omega.Dot(face.normal_);
+                if (mu < 0.0)
+                {
+                  const auto& wt = quadrature->weights_[n];
+                  const auto weight = -mu * wt * intF_shapeI;
+                  const auto dof_map = discretization.MapDOFLocal(cell, i, uk_man, n, 0);
+
+                  for (size_t gsg = 0; gsg < num_gs_groups; ++gsg)
+                    local_response +=
+                      weight * psi_dagger[gs][dof_map + gsg] * psi_bndry[num_gs_groups * n + gsg];
+                } // if outgoing
+              }
+            } // for face node fi
+          }
+          ++f;
+        } // for face
+      }   // for cell
+      ++gs;
+    } // for groupset
+  }   // if boundary sources
```

Comment:
same. should you remove 2 white spaces here to move the comment a tad to the left?